src/Grpc.AspNetCore.Server/ServerMethods.cs

```diff
@@ -0,0 +1,33 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Threading.Tasks;
+using Grpc.Core;
+
+namespace Grpc.AspNetCore.Server
+{
+    // We're using an open delegate (the first argument is the TService instance) to represent calls here since the instance is create per request.
+
+    public delegate Task<TResponse> UnaryServerMethod<TRequest, TResponse, TService>(TService service, TRequest request, ServerCallContext serverCallContext);
+
+    public delegate Task<TResponse> ClientStreamingServerMethod<TRequest, TResponse, TService>(TService service, IAsyncStreamReader<TRequest> stream, ServerCallContext serverCallContext);
+
+    public delegate Task ServerStreamingServerMethod<TRequest, TResponse, TService>(TService service, TRequest request, IServerStreamWriter<TResponse> stream, ServerCallContext serverCallContext);
+
+    public delegate Task DuplexStreamingServerMethod<TRequest, TResponse, TService>(TService service, IAsyncStreamReader<TRequest> input, IServerStreamWriter<TResponse> output, ServerCallContext serverCallContext);
```

Comment:
The open delegates are now public because they are created by the invoker factory. Thoughts?

Reply:
If we make them public I'd prefer for them to live in Grpc.Core

Reply:
Maybe. Nothing in Grpc.Core.Api would use them unless codegen and ServiceBinderBase was updated.

Grpc.Core.* is locked down for 1.19 so they have to live here for the first preview.

Reply:
Why does the invoker factory need to be public?

Reply:
I could be internal and be accessed via InternalVisibleTo. It could be useful for other devs who don't want to be died to reflection to get the call handler method.

Reply:
We should only make things like that public when people ask. It’s looking pretty niche right now

Reply:
I'd strongly advise against making implementation details like this public. If users ask for it, consider making a *different thing* for their scenario and making that public instead.